WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -981,3 +985,66 @@ private extension InsightStoreState {
     }
 
 }
+
+// MARK: - iOS 14 Widgets Data
+private extension InsightStoreState {
+
+    private func saveHomeWidgetTodayData(stats: TodayWidgetStats) {
+
+        guard let siteID = SiteStatsInformation.sharedInstance.siteID else {
+            return
+        }
+
+        var homeWidgetTodayCache = HomeWidgetTodayData.read() ?? initializeHomeWidgetTodayData()
+
+        guard let oldData = homeWidgetTodayCache[siteID.intValue] else {
+            DDLogError("HomeWidgetToday: Failed to find a matching site")
+            return
+        }
+        // TODO - TODAYWIDGET: it might be better to move the blog updates in SiteStatsInformation
+        let blogService = BlogService(managedObjectContext: ContextManager.shared.mainContext)
+
+        guard let blog = blogService.blog(byBlogId: siteID) else {
+            DDLogError("HomeWidgetToday: the site does not exist anymore")
+            // if for any reason that site does not exist anymore, remove it from the cache.
+            homeWidgetTodayCache.removeValue(forKey: siteID.intValue)
+            HomeWidgetTodayData.write(data: homeWidgetTodayCache)
+            return
+        }
+        // refresh stats and update any blog info, if they had changed
+        homeWidgetTodayCache[siteID.intValue] = HomeWidgetTodayData(siteID: siteID.intValue,
+                                                           siteName: blog.title ?? oldData.siteName,
+                                                           url: blog.url ?? oldData.url,
+                                                           timeZoneName: blogService.timeZone(for: blog).identifier,
+                                                           date: Date(),
+                                                           stats: stats)
+
+        HomeWidgetTodayData.write(data: homeWidgetTodayCache)
+        if #available(iOS 14.0, *) {
+            WidgetCenter.shared.reloadTimelines(ofKind: "WordPressHomeWidgetToday")
```

Comment:
Should we have a constant for `"WordPressHomeWidgetToday"`?

Reply:
Or alternatively this could be in a constants file shared between the two targets (since this value is also used in the widget code).

Not sure if that'd be good practice or bad practice to share a file like that across projects, though.  It's just a quick thought I had.

Reply:
I'm not a fan of that either, but since we already have that ('Constants.h`, `Constants.m`) I'm just gonna put it in there